### PR TITLE
Add source filter to front page

### PR DESCRIPTION
**Pull request by stefanv**
Main new features of this PR:
- Add front end source source list filtering capability
- Handle null RA/Dec in front-end source list display
- Limit front end source list display to 100 per page and add pagination functionality
- Re-structure front-page components
- Use arrow in place of datetime to facilitate generating raw SQL strings (for future mat view PR)

Continues work from https://github.com/skyportal/skyportal/pull/104

Remaining items:

- [x] Reduce number of data files (currently 14MB)

### CODE REVIEW

File: skyportal/handlers/source.py [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
+                                                   joinedload(Source.thumbnails)
+                                                   .joinedload(Thumbnail.photometry)
+                                                   .joinedload(Photometry.instrument)
+                                                   .joinedload(Instrument.telescope)])
+        elif page_number_given:
+            page = int(source_id_or_page_num)
+            q = Source.query.filter(Source.id.in_(DBSession.query(
+                GroupSource.source_id).filter(GroupSource.group_id.in_(
+                    [g.id for g in self.current_user.groups]))))
+            sql_str = str(q.statement.compile(dialect=postgresql.dialect(),
```
Comment (stefanv) [inline skyportal/handlers/source.py]: Still needed?

File: skyportal/handlers/source.py [hunk 2]
```diff
@@ -64,3 +88,52 @@ def delete(self, source_id):
         DBSession().commit()
 
         return self.success(action='cesium/FETCH_SOURCES')
+
+
+class FilterSourcesHandler(BaseHandler):
+    @auth_or_token
+    # def get(self, sourceID=None, ra=None, dec=None, radius=None,
```
Comment (stefanv) [inline skyportal/handlers/source.py]: Old?

File: skyportal/plot.py [hunk 3]
```diff
@@ -159,35 +160,61 @@ def photometry_plot(source_id):
...
-                       .query(Photometry, Telescope.nickname.label('telescope'))
-                       .join(Instrument).join(Telescope)
+                       .query(Photometry, Telescope.nickname.label('telescope'), Source.varstar)
+                       .join(Instrument).join(Telescope).join(Source)
                        .filter(Photometry.source_id == source_id)
                        .statement, DBSession().bind)
     if data.empty:
         return None, None, None
 
+    for col in data:
```
Comment (stefanv) [inline skyportal/plot.py]: Are these plotting changes related to this PR?

File: requirements.txt [hunk 4]
```diff
@@ -7,3 +7,11 @@ bokeh==0.12.9
 pytest-randomly==2.1.1
 factory-boy==2.8.1
 astropy==3.0.3
+tornado==4.5.3
+aplpy==1.1.1
```
Comment (stefanv) [inline requirements.txt]: All needed?

File: skyportal/plot.py (hunk 3 above)
Comment (acrellin) [reply skyportal/plot.py]: No, these aren't my changes - this is part of what I was referring to when I was saying the merge/rebase looked botched...

File: requirements.txt (hunk 4 above)
Comment (acrellin) [reply requirements.txt]: yep

**Comment by stefanv** [comment]
Hrm, I see. I wonder why they are showing up. It seems like it was indeed 
botched. Can you scan the history and remove them?

**Comment by acrellin** [comment]
Ready for final review, @stefanv

**Comment by stefanv** [comment]
Looks good, Ari!

Checks:
- How much of the functionality is covered by new tests?
- The plotting changes are still in there; should we remove those for this PR?  They do strange things like reassign bands.
- The much bigger database table: are these fields all fairly generic?  I recall @dannygoldstein saying perhaps not.  Thoughts, Danny?

**Comment by dannygoldstein** [comment]
Hi all,

The new columns in source are fairly general, but I worry that eventually the Source table will grow to so many columns that it will be difficult to keep track of the schema. To simplify things, I wonder if it would be better to have a single metadata column with the `JSONB` datatype, and all columns such as the following could be fields in a JSONB document (no-sql style):

    simbad_class = sa.Column(sa.Unicode, nullable=True, )
    simbad_info = sa.Column(JSONB, nullable=True)
    gaia_info = sa.Column(JSONB, nullable=True)
    tns_info = sa.Column(JSONB, nullable=True)
    tns_name = sa.Column(sa.Unicode, nullable=True)

Would this add on unnecessary overhead?

**Comment by stefanv** [comment]
I think this is the right way to go; we cannot add columns customized to each survey.

**Comment by acrellin** [comment]
@stefanv I've addressed all of your feedback, including removing the unrelated changes to `plot.py`.  Ready for you to merge!

**Comment by stefanv** [comment]
Some conflicts; otherwise looks good to go.

File: skyportal/tests/tools/test_ztf_upload.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+fname = "skyportal/tests/data/541234765015015012.avro"
+
+# this is a stub. Fill it in.
+def test_ztf_upload(public_group, user, capsys):
+
+    with capsys.disabled():
+        print(user)
+
+    #a = ztf_upload_avro.ZTFAvro(fname, z, clobber=True)
```
[the remark this answers is not in the document]
Comment (stefanv) [reply skyportal/tests/tools/test_ztf_upload.py]: And this incomplete stub... remove?

[the remark this answers is not in the document]
Comment (stefanv) [reply skyportal/tests/tools/test_ztf_upload.py]: Or complete; latter would be preferable, I think.

**Comment by acrellin** [comment]
@stefanv conflicts resolved

**Comment by stefanv** [comment]
Only outstanding issue here is `skyportal/tests/tools/test_ztf_upload.py`

**Comment by acrellin** [comment]
You said to pretend that doesn't exist, remember? 😉

**Comment by stefanv** [comment]
I did say that, didn't I :)

File: requirements.txt (hunk 4 above)
Comment (stefanv) [reply requirements.txt]: Why is tornado being removed?

Comment (stefanv) [reply requirements.txt]: Ah, baselayer requirement?

**Comment by acrellin** [comment]
Exactly - it's already in baselayer.

File: requirements.txt (hunk 4 above)
Comment (acrellin) [reply requirements.txt]: Exactly - it's already in baselayer.

**Comment by stefanv** [comment]
:clap: This was a big one; thank you for shepherding this through, @acrellin!
